handlers/http.go

```diff
@@ -105,27 +128,45 @@ func (handler *HTTPClientHandler) Handle(event *types.Event, conf *types.Handler
 func (handler *HTTPClientHandler) httpDo(conf *types.HandlerConfig) (*http.Response, error) {
 	buff := bytes.NewBuffer([]byte(conf.Body))
 
-	discoveredURI, err := handler.resolv.Discover(conf.URI)
-	if err != nil {
-		log.Printf("[ERROR] Discovering URI [%s]: %s\n", conf.URI, err.Error())
-		log.Println("[DEBUG] Will be used system DNS server")
-	} else {
-		conf.URI = discoveredURI
-	}
+	var response *http.Response
+	var err error
+
+	attempt := 1
+	for {
+		discoveredURI, err := handler.resolv.Discover(conf.URI)
+		if err != nil {
+			log.Printf("[ERROR] Discovering URI [%s]: %s\n", conf.URI, err.Error())
+			log.Println("[DEBUG] Will be used system DNS server")
+		} else {
+			conf.URI = discoveredURI
+		}
+
+		req, err := http.NewRequest(handler.conf.Method, conf.URI, buff)
+		if err != nil {
+			return nil, err
+		}
 
-	req, err := http.NewRequest(handler.conf.Method, conf.URI, buff)
-	if err == nil {
 		if handler.conf.Headers != nil {
 			for k, v := range handler.conf.Headers {
 				req.Header.Set(k, v)
 			}
 		}
 
 		log.Printf("[DEBUG] handler=http uri='%s' body='%s'", conf.URI, conf.Body)
-		return handler.client.Do(req)
+		response, err = handler.client.Do(req)
+		if err == nil && response.StatusCode == 200{
+			break
+		}
+
+		if attempt >= handler.retries {
+			break
+		}
+
+		attempt ++
+		time.Sleep(handler.backoff.Next(attempt))
```

Comment:
Hi. Thanks for the comment. I have fix the problem with attempt incrementation before wait. Also I have added Constant backoff(waiting the same interval before attempts) and made it by default. I didn't remove Linear backoff, I think in some cases it can be useful, so now user can specify backoff strategy in options.